test/gurka/test_landmarks.cc

```diff
@@ -142,3 +141,101 @@ TEST(LandmarkTest, TestParseLandmarks) {
+  uint32_t edge_index = 0;
+
+  // either test the fixed landmark or flexible landmarks as following
+  // double lng = 0, lat = 0;
+  // const Landmark landmark_fixed(1, "fixed landmark", LandmarkType::casino, lng, lat);
+
+  for (const auto& e : tb.directededges()) {
+    std::vector<PointLL> shape = tb.edgeinfo(&e).shape();
+    auto point = shape[shape.size() / 2];
+    auto ltype = static_cast<LandmarkType>((edge_index + 1) % invalid_landmark);
+
+    Landmark landmark(edge_index, std::to_string(edge_index), ltype, point.first, point.second);
+
+    auto edge_id = tile_id;
+    edge_id.set_id(edge_index++);
+
+    // tb.AddLandmark(edge_id, landmark_fixed);
+    tb.AddLandmark(edge_id, landmark);
+  }
+
+  // call the store graphtile function to overwrite the tile on disk with the new info
+  tb.StoreTileData();
+
+  // instantiate a graphreader using the config
+  GraphReader gr(landmark_map.config.get_child("mjolnir"));
+
+  // call getgraphtile on it
+  auto tile = gr.GetGraphTile(tile_id);
+
+  // get the edgeinfo using the edge who you added the landmark to
+  // call GetNamesAndTypes on the edgeinfo
+  // loop over the results until the type is kLandmark when it is then you need to use the method
+  // to decode the string into a landmark object. asser that the landmark you got out matches the one
+  // you told it to add (excepting the id because we dont store that).
+
+  for (const auto& e : tile->GetDirectedEdges()) {
+    auto ei = tile->edgeinfo(&e);
+    auto tagged_values = ei.GetTags();
+
+    for (const auto& value : tagged_values) {
+      if (value.first != baldr::TaggedValue::kLandmark)
+        continue;
+
+      Landmark landmark(value.second);
+      std::cout << "landmark: " << landmark.id << " " << landmark.name << " "
+                << static_cast<int>(landmark.type) << " " << landmark.lng << " " << landmark.lat
+                << std::endl;
+
+      // check data correctness of the fixed landmark
+      // EXPECT_EQ(landmark.id, 0);
+      // EXPECT_EQ(landmark.name, "fixed landmark");
+      // EXPECT_EQ(landmark.type, LandmarkType::casino);
+      // EXPECT_EQ(landmark.lng, lng);
+      // EXPECT_EQ(landmark.lat, lat);
+    }
+
+    auto values = ei.GetTaggedValues();
+    for (const std::string& v : values) {
+      if (static_cast<baldr::TaggedValue>(v[0]) != baldr::TaggedValue::kLandmark) {
+        continue;
+      }
+
+      Landmark landmark(v.substr(1));
+      std::cout << "landmark: " << landmark.id << " " << landmark.name << " "
+                << static_cast<int>(landmark.type) << " " << landmark.lng << " " << landmark.lat
+                << std::endl;
+
+      // check data correctness of the fixed landmark
+      // EXPECT_EQ(landmark.id, 0);
+      // EXPECT_EQ(landmark.name, "fixed landmark");
+      // EXPECT_EQ(landmark.type, LandmarkType::casino);
+      // EXPECT_EQ(landmark.lng, lng);
+      // EXPECT_EQ(landmark.lat, lat);
+    }
```

Comment:
this needs to be finished, printing doesnt count as tests ;)

Reply:
ok will get it done